activerecord/lib/active_record/connection_adapters/postgresql/oid/range.rb

```diff
@@ -69,13 +69,24 @@ def type_cast_single_for_database(value)
             def extract_bounds(value)
               from, to = value[1..-2].split(",", 2)
               {
-                from:          (from == "" || from == "-infinity") ? infinity(negative: true) : from,
-                to:            (to == "" || to == "infinity") ? infinity : to,
+                from:          (from == "" || from == "-infinity") ? infinity(negative: true) : unquote(from),
+                to:            (to == "" || to == "infinity") ? infinity : unquote(to),
                 exclude_start: value.start_with?("("),
                 exclude_end:   value.end_with?(")")
               }
             end
 
+            def unquote(value)
+              if value[0] == '"' && value[-1] == '"'
```

Comment:
`value[0]` and `value[-1]` allocates a new string, use non-allocation `start_with?` and `end_with?`.

```suggestion
              if value.start_with?('"') && value.end_with?('"')
```